packages/datadog-instrumentations/src/moleculer/client.js

```diff
@@ -9,29 +9,28 @@ const errorChannel = channel('apm:moleculer:call:error')
 
 function wrapCall (call) {
   return function (actionName, params, opts) {
-    const callResource = new AsyncResource('bound-anonymous-fn')
-
     opts = arguments[2] = opts || {}
     opts.meta = opts.meta || {}
 
     arguments.length = Math.max(3, arguments.length)
 
-    return callResource.runInAsyncScope(() => {
-      startChannel.publish({ actionName, params, opts })
-
+    const ctx = { actionName, params, opts }
+    return startChannel.runStores(ctx, () => {
       const promise = call.apply(this, arguments)
       const broker = this
-      const ctx = promise.ctx
+      ctx.promiseCtx = promise.ctx
```

Comment:
It would be better to pass what we need directly here, but that's a nit and could be cleaned up later if needed so not blocking on this.